File: day1/time-conversion/solution.py
s = '12:05:45AM';

def timeConversion(s):
    splited = list(s)
    hora = int(''.join(splited[0:2]))
    ampm = splited[-2]

    if(ampm == 'A'):
        if(hora == 12):
            hora = 0
        if(hora < 10):
            hora = f'0{hora}'
    if(ampm == 'P'):
        if hora < 12:
            hora = hora+12
    result = ''.join(splited[2:-2])        
    print (f'{hora}{result}')
    return f'{hora}{result}'
timeConversion(s)   

File: day1/time-conversion/solution.js
let s = '01:05:45AM';

function timeConversion(s) {
  let splited = s.split('');
  let hora = Number(`${splited[0]}${splited[1]}`);
  let ampm = splited[splited.length - 2];

  if (ampm == 'A') {
    if (hora == 12) {
      hora = '00';
    } else {
      hora = hora.toLocaleString('pt-BR', {
        minimumIntegerDigits: 2
      });
    }
  }
  if (ampm == 'P') {
    if (hora < 12) {
      hora = hora + 12;
    }
  }
  return `${hora}${s.substring(2, 8)}`;
}

console.log(timeConversion(s));
